osmoutils/coin_helper.go

```diff
@@ -25,3 +29,16 @@ func MinCoins(coinsA sdk.Coins, coinsB sdk.Coins) sdk.Coins {
 	}
 	return resCoins
 }
+
+func SubDecCoinArrays(decCoinsArrayA []sdk.DecCoins, decCoinsArrayB []sdk.DecCoins) ([]sdk.DecCoins, error) {
+	if len(decCoinsArrayA) != len(decCoinsArrayB) {
+		return []sdk.DecCoins{}, fmt.Errorf("DecCoin arrays must be of equal length to be subtracted")
+	}
+
+	finalDecCoinArray := []sdk.DecCoins{}
+	for i := range decCoinsArrayA {
+		finalDecCoinArray = append(finalDecCoinArray, decCoinsArrayA[i].Sub(decCoinsArrayB[i]))
+	}
+
+	return finalDecCoinArray, nil
+}
```

Comment:
Can we add a godoc? Also, do you think we should:

* sort the two coinArrays prior to subtracting them
* check that the denom at each index matches
* add tests for this helper

Reply:
Good point, will add a godoc.

As for your other points:
> sort the two coinArrays prior to subtracting them

I don't think there's a clear dimension along which we can sort them at this level of abstraction. It would also make the function much less generally useful imo. All uptime-related calls for this function are already sorted by construction so this wouldn't make a difference on that front.

> check that the denom at each index matches

This func actually doesn't deal with denoms at all – each index of the array is a `DecCoins` so we defer to `DecCoins`'s `Sub` function which already handles all denom-related edge cases.

> add tests for this helper

I figured this was a basic enough function that's already implicitly tested where it's used, but I agree it would be good practice to add tests for this (and the other helpers in this file)